### PR TITLE
docs(product): add docs for using the InMemory driver

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Using the InMemory Product driver is a bit difficult, because there's no documentation.

Fixes: N/A

## What is the new behavior?
Added documentation.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/README.md [hunk 1]
```diff
@@ -9,3 +9,58 @@ provides clear interfaces, models, and factories for the frontend of an ecommerc
...
+}
+
+export interface MockMyAppDatabase {
+  products: DaffProduct[];
+  myAppCollection: MyAppCollection[];
+}
+```
+With this, you'll have all get requests for products handled by `@daffodil/product` and all other requests handled by your `MyAppTestingService`.
+
+The only other thing you need to do is to make the product images in `@daffodil/product/assets` available to your application, because these are the image references used in the mocked product data.
```
Comment (damienwebdev) [inline libs/product/README.md]: How do I do this?

File: libs/product/README.md [hunk 2]
```diff
@@ -9,3 +9,58 @@ provides clear interfaces, models, and factories for the frontend of an ecommerc
 ```
 npm install @daffodil/product
 ```
+
+## Using the InMemory Driver
```
Comment (damienwebdev) [inline libs/product/README.md]: I don't think we should ever assume a user will only use JUST the product in-mem. While it is possible, its also too confusing to explain the difference between one and the other. We should tend towards "singular" configuration policies vs. multiple.

File: libs/product/README.md [hunk 3]
```diff
@@ -9,3 +9,58 @@ provides clear interfaces, models, and factories for the frontend of an ecommerc
...
+
+The InMemory driver is for rapid development without the need to set up a magento/shopify/etc backend. It will mock out the service calls to get products and return fake data. There are a few steps to wiring up the DaffInMemory driver:
+
+If you only plan on making serving calls for getting products, you can simply import `HttpClientInMemoryWebApiModule.forRoot(DaffInMemoryBackendProductService)` and `DaffProductInMemoryDriverModule.forRoot()` into your AppModule, and it will just work. If you need to make service calls that involve a cart or anything else, you'll need to create a middle layer between your app and the `DaffProductInMemoryDriverModule` so that the `DaffInMemoryBackendProductService` only ever receives product-related requests. This middle layer can just be an `in-memory-web-api` service that uses the `DaffInMemoryBackendProductService` to handle database creation for products and product get requests:
+
+```
```
Comment (damienwebdev) [inline libs/product/README.md]: Add the language hint.

File: libs/product/README.md [hunk 4]
```diff
@@ -9,3 +9,58 @@ provides clear interfaces, models, and factories for the frontend of an ecommerc
 ```
 npm install @daffodil/product
 ```
+
```
Comment (damienwebdev) [inline libs/product/README.md]: I would move this into a separate "stories" folder.

File: libs/product/README.md (hunk 3 above)
Comment (lderrickable) [reply libs/product/README.md]: What's the language hint?

File: libs/product/README.md (hunk 1 above)
Comment (lderrickable) [reply libs/product/README.md]: added some extra stuff and an example with angular-cli

File: libs/product/README.md (hunk 2 above)
Comment (lderrickable) [reply libs/product/README.md]: changed to just have the more complex version.

Comment (lderrickable) [reply libs/product/README.md]: done

File: libs/product/README.md (hunk 3 above)
Comment (damienwebdev) [reply libs/product/README.md]: You can add ```ts to colorize a codeblock in the TypeScript language format.

File: libs/product/README.md (hunk 4 above)
Comment (damienwebdev) [reply libs/product/README.md]: To clarify, I don't think this should be in the README of `product`. I think we should have a separate `stories` folder for docs which describes a "story" a developer would undergo.

File: libs/product/README.md (hunk 3 above)
Comment (lderrickable) [reply libs/product/README.md]: done

File: libs/product/README.md (hunk 4 above)
Comment (lderrickable) [reply libs/product/README.md]: done

**Pull request merged**